dynamiq/memory/backends/dynamo_db.py

```diff
@@ -0,0 +1,421 @@
+import json
+import time
+import uuid
+from decimal import Decimal, InvalidOperation
+from typing import Any
+
+from boto3.dynamodb.conditions import Attr
+from botocore.exceptions import ClientError
+from pydantic import ConfigDict, Field, PrivateAttr
+
+from dynamiq.connections import AWS
+from dynamiq.memory.backends.base import MemoryBackend
+from dynamiq.prompts import Message, MessageRole
+from dynamiq.utils.logger import logger
+
+
+class DynamoDBMemoryError(Exception):
+    """Base exception class for DynamoDB Memory Backend errors."""
+
+    pass
+
+
+def _convert_floats_to_decimals(obj):
```

Comment:
- Please add annotations
- please check for https://docs.python.org/3.12/library/json.html `parse_float` for potencial usage. It seems this a pretty common solution for DynamoDB